### DISCUSSION TITLE
[需求/求助]关于postFinder.list({...})分页

### DISCUSSION THREAD

**Discussion by AR-26710** [halo-dev/halo#6965] [Q&A]
### 你当前使用的版本

2.20.7

### 描述一下此特性

叠甲：新手小白

在首页添加最多观看、最近更新后在主题开发文档Finder API中没找到切换分页功能，即上下一页链接，故无法使用分页查看后面的内容，只能显示page: 1,size数量的文章，size为后台默认数量，size的默认数量似乎是后台设置中的**文章列表显示条数**。
``` html
# 最多观看
<div class="content" id="tab2" style="display: none;">
    <ul class="joe_list">
          <th:block th:with="posts = ${postFinder.list({page: 1, sort: {'stats.visit,desc', 'metadata.creationTimestamp,asc'}})}">
                  <th:block th:each="post, iteration : ${posts.items}">
                          <th:block th:replace="~{modules/macro/post_item :: post_item(type = 'hot')}" />
                    </th:block>
           </th:block>
    </ul>
</div>
# 最近更新
<div class="content" id="tab3" style="display: none;">
      <ul class="joe_list">
              <th:block th:with="posts = ${postFinder.list({page: 1, sort: {'status.lastModifyTime,desc', 'metadata.creationTimestamp,asc'}})}">
                      <th:block th:each="post, iteration : ${posts.items}">
                              <th:block th:replace="~{modules/macro/post_item :: post_item(type = 'last')}" />
                      </th:block>
              </th:block>
      </ul>
</div>
```

在主题开发模板变量中可以使用`nextUrl` `prevUrl`来进行上下页码的切换，且有专属的请求路径。
[UrlContextListResult<ListedPostVo>](https://docs.halo.run/developer-guide/theme/template-variables/index_#urlcontextlistresultlistedpostvo)
![Image](https://github.com/user-attachments/assets/1ae0c389-7782-49f4-8531-f651c12da578)

但在Finder API中虽然给了分页参数但没有相应的上下一页链接，没有请求路径，所以似乎无论是切换页码还是向下加载都需要使用相关api与ajax进行请求，但不使用api而使用Finder API不就是为了方便与减少api的暴露吗。
[ListResult<ListedPostVo>](https://docs.halo.run/developer-guide/theme/finder-apis/post#listresultlistedpostvo)
![Image](https://github.com/user-attachments/assets/98b9f6ac-9e2e-4f05-bb95-901d490c4f5a)

如果理解有误，请指正或提供思路。
如没有希望`postFinder.list({...})`后续能支持主动切换页码功能。

**Comment by JohnNiang** [comment]
Hi @AR-26710 , thank you for reaching out here!

因为 postFinder#list 方法可以在任意模板中调用，所以我们没有办法知道 Base URL 到底是什么，所以无法构建出 nextURL 和 prevURL。在是该方法返回的 ListResult 中已经包含了 page、hasNext 和 hasPrevious 参数，满足自行构建上一页和下一页的条件。

另外，应该是可以直接传入 size 参数来控制查询条数。

/kind support

**Comment by AR-26710** [comment]
ok，谢谢。
